### PR TITLE
Refactor gpcheckcat to modularize the repair code and add repair for the missing_extraneous module.

### PR DESCRIPTION
**Pull request by kaknikhil**
1. Moved all the generic repair code to its own module.
2. New module for missing_extraneous repair.
3. Make gpcheckcat do repairs per check instead of doing it after all
the checks have finished
4. Fix gpcheckcat so that it now creates a repair dir with -R option.
5. Unit and behave tests for the above mentioned modules.
6. Update gpcheckcat behave to use consistent db names
7. Behave tests for a few other checks like foreign_key, constraint_db,
part_integrity and also for the -g option

### CODE REVIEW

**Comment by ivannovick** [comment]
Very cool

File: gpMgmt/bin/gpcheckcat_modules/repair.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+import os
+import stat
+from datetime import datetime
+from gpcheckcat_modules.repair_missing_extraneous import RepairMissingExtraneous
```
Comment (asimrp) [inline gpMgmt/bin/gpcheckcat_modules/repair.py]: Is there a reason `RepairMissingExtraneous` is written in a separate Python module and not in `repair.py`?

File: gpMgmt/bin/gpcheckcat_modules/repair.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        c = self._context.report_cfg[segment_id]
+        out_filename = "{0}_{1}_{2}".format(self._context.dbname, self._issue_type, self.TIMESTAMP)
+        bash_script_content = '\necho "{0}"\n'.format(self._desc)
+        bash_script_content += self.__get_psql_command(segment_id).format(hostname=c['hostname'],
+                                                                          port=c['port'], sql=filename,
+                                                                          dbname=self._context.dbname,
+                                                                          filename=out_filename)
+        return bash_script_content
+
+    def __get_psql_command(self, segment_id):
```
Comment (asimrp) [inline gpMgmt/bin/gpcheckcat_modules/repair.py]: `class Psql` in `gpMgmt/bin/gppylib/commands/gp.py` may be reused here, can you please check?

File: gpMgmt/bin/gpcheckcat_modules/repair.py (hunk 1 above)
Comment (kaknikhil) [inline gpMgmt/bin/gpcheckcat_modules/repair.py]: repair.py is not responsible for generating the repair content. The responsibility of this class is to ask for repair contents and then create the repair dir and the corresponding sql and bash scripts.

File: gpMgmt/bin/gpcheckcat_modules/repair.py (hunk 2 above)
Comment (kaknikhil) [inline gpMgmt/bin/gpcheckcat_modules/repair.py]: this was borrowed from the old code . i will take a look at gp.py and reuse it if possible , thanks for the suggestion

File: gpMgmt/bin/gpcheckcat_modules/repair.py (hunk 1 above)
Comment (Chibin) [inline gpMgmt/bin/gpcheckcat_modules/repair.py]: Probably add some description of the purpose of repair.py at the top?

Comment (kaknikhil) [inline gpMgmt/bin/gpcheckcat_modules/repair.py]: yes i agree. pushed a commit for it.

File: gpMgmt/bin/gpcheckcat_modules/repair.py (hunk 2 above)
Comment (kaknikhil) [inline gpMgmt/bin/gpcheckcat_modules/repair.py]: @asimrp : i looked into the Psql class, we will need to make significant changes(read hacks) to it for us to be able to use it in gpcheckcat. so we decide not to reuse it here.

**Pull request merged**